File: demo/input/Greeter.ts
namespace MyTest {
    "use strict";

    export class Greeter implements IGreeter {
        private _useName: string;
        private _greeting: string;

        constructor() {
            this._useName = '';
            this._greeting = '';
        }

        public userName(value?: string): string {
            if (arguments.length== 0) {
                return this._useName;
            } else {
                this._useName = value;
            }
        }

        public get greeting(): string {
            return this._greeting;
        }

        public set greeting(value: string) {
            this._greeting = value;
        }

        greet(): string {
            return "Hello, " + this.greeting;
        }
    }
}

File: demo/csharp/Greeter.cs
using System.Linq;
using TypeScript.CSharp;

namespace Demo
{
    public class Greeter : Object, IGreeter
    {
        private String _useName;
        private String _greeting;
        public Greeter()
        {
            this._useName = "";
            this._greeting = "";
        }

        public String userName()
        {
            return this._useName;
        }

        public void userName(String value)
        {
            this._useName = value;
        }

        public String greeting
        {
            get
            {
                return this._greeting;
            }

            set
            {
                this._greeting = value;
            }
        }

        public String greet()
        {
            return "Hello, " + this.greeting;
        }
    }
}